code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -610,6 +614,56 @@ func (sa *StorageAllocation) restMinLockDemand() (rest state.Balance) {
 	return
 }
 
+func (sa *StorageAllocation) addWritePoolOwner(userId string) {
+	for _, id := range sa.WritePoolOwners {
```

Comment:
It might have `nil pointer dereference` crash too.

Reply:
As before, if `sa` is nil then there is a serious coding error elsewhere and the chain is going to crash in the next few nanoseconds anyway.